airflow/jobs/local_task_job.py

```diff
@@ -291,3 +293,19 @@ def _update_dagrun_state_for_paused_dag(self, session=None):
             if dag_run:
                 dag_run.dag = dag
                 dag_run.update_state(session=session, execute_callbacks=True)
+
+    @staticmethod
+    def _enable_task_listeners():
+        """
+        Check if we have any registered listeners, then register sqlalchemy hooks for
+        TI state change if we do.
+        """
+        from airflow.plugins_manager import integrate_listener_plugins
+
+        integrate_listener_plugins()
+        from airflow.listeners.listener import get_listener_manager
+
+        if get_listener_manager().has_listeners():
+            from airflow.listeners.events import register_task_instance_state_events
```

Comment:
Is there a reason why imports are in such order, if yes, can we add comments explaining that please

Reply:
I'm only importing `register_task_instance_state_events` now.